homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -216,32 +215,25 @@ def __init__(
         self._station = sensor_station
         self._weather_api = weather_api
         self._weather: WeatherStationInfo | None = None
-        self._active: bool | None = None
-        self._measure_time: str | None = None
-
-    @property
-    def extra_state_attributes(self) -> dict[str, Any]:
-        """Return the state attributes of Trafikverket Weatherstation."""
-        _additional_attributes: dict[str, Any] = {
-            ATTR_ATTRIBUTION: ATTRIBUTION,
-        }
-        if self._active:
-            _additional_attributes[ATTR_ACTIVE] = self._active
-        if self._measure_time:
-            _additional_attributes[ATTR_MEASURE_TIME] = self._measure_time
-
-        return _additional_attributes
 
     @Throttle(MIN_TIME_BETWEEN_UPDATES)
     async def async_update(self) -> None:
         """Get the latest data from Trafikverket and updates the states."""
         try:
             self._weather = await self._weather_api.async_get_weather(self._station)
-            self._attr_native_value = getattr(
-                self._weather, self.entity_description.api_key
-            )
         except (asyncio.TimeoutError, aiohttp.ClientError, ValueError) as error:
             _LOGGER.error("Could not fetch weather data: %s", error)
             return
-        self._active = self._weather.active
-        self._measure_time = self._weather.measure_time
+        self._attr_native_value = getattr(
+            self._weather, self.entity_description.api_key
+        )
+        if (
+            self._attr_native_value is None
+            and self.entity_description.state_class == SensorStateClass.MEASUREMENT
```

Comment:
We use `None` to represent an unknown state.  Is this really 0 or should it be unknown?

If it is zero, this would be a bit cleaner if it took a set of keys to match against instead of using the state class.

```self.entity_description.key in NONE_IS_ZERO_SENSORS```

Reply:
It's a very strange way of doing it from the api but to give an example if it's not raining it's not reporting 0 precipitation, instead it doesn't report anything e.g. it comes back as None. Therefore I need to have this to report 0 instead of None as the state is not unknown.
Your suggestion looks good for me to implement and makes it easy to understand. Will fix